### PR TITLE
SUS-4573 | Run MediaWiki in Docker locally

### PR DESCRIPTION
**Pull request by dmnsobczak**
* use `php-fpm` and `nginx` duo
* use a local instance of `memcached` to greatly speed up the Docker instance running on 127.0.0.1

### CODE REVIEW

File: docker/dev/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on:
+      - php-wikia
+  php-wikia:
+    image: php-wikia-dev:latest
+    ports:
+      - "9000:9000"
+    volumes:
+      - ./../../../app:/php-wikia-dev/src
+      - ./../../../config:/php-wikia-dev/config
+      - ./../../../cache/messages:/php-wikia-dev/cache/messages
```
Comment (macbre) [inline docker/dev/docker-compose.yml]: Let's keep these consistent with:

```
 VOLUME /usr/wikia/slot1/current/src
 VOLUME /usr/wikia/slot1/current/config
 VOLUME /usr/wikia/slot1/current/cache/messages
```

File: docker/base/Dockerfile [hunk 2]
```diff
@@ -12,6 +12,7 @@ RUN apt-get update && apt-get install -y \
     libbz2-dev \
     libjpeg-dev \
     libpng-dev \
+    libxml2-dev \
```
Comment (macbre) [inline docker/base/Dockerfile:15]: @mszabo-wikia - why do we need simplexml?

Comment (mszabo-wikia) [reply docker/base/Dockerfile:15]: Needed by Design System

Comment (macbre) [reply docker/base/Dockerfile:15]: `SimpleXMLElement` right?

Comment (mszabo-wikia) [reply docker/base/Dockerfile:15]: Yup

File: docker/base/Dockerfile [hunk 3]
```diff
@@ -71,7 +69,11 @@ RUN docker-php-ext-install \
     bz2 \
     gd \
     mysqli \
-    opcache
+    opcache \
+    # simplexml: SimpleXMLElement is used by DesignSystem and PortableInfoboxBuilder extensions
+    simplexml \
+    # needed by RabbitMQ client
+    bcmath
```
Comment (macbre) [inline docker/base/Dockerfile:76]: :+1:

File: docker/dev/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  php-wikia:
+    image: php-wikia-dev
+    volumes:
+      - ../../../app:/usr/wikia/slot1/current/src
+      - ../../../config:/usr/wikia/slot1/current/config
+      - ../../../cache:/usr/wikia/slot1/current/cache/messages
+    depends_on:
+      - memcached
+    environment:
+      - MEMCACHED_SERVER=memcached:11211
```
Comment (macbre) [inline docker/dev/docker-compose.yml:21]: See https://github.com/Wikia/config/pull/2458

**Pull request merged**